### PR TITLE
ERC-4626

### LINKED ISSUE #511: Reference ERC4626 Implementation
Add a subdirectory to the `examples/` that indexes erc4626.

### PR DESCRIPTION
**Pull request by kdonthi**
Closes #511

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -3846,6 +3875,18 @@ packages:
       - zod
     dev: true
 
+  /@safe-global/safe-apps-provider@0.17.1(typescript@5.3.3):
```
Comment (kyscott18) [inline pnpm-lock.yaml:3878]: How did this get added to the lockfile?

File: examples/reference-erc4626/ponder.config.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { createConfig } from "@ponder/core";
+import { erc4626ABI } from "@wagmi/core";
```
Comment (kyscott18) [inline examples/reference-erc4626/ponder.config.ts:2]: Nice using the correct abi

File: examples/reference-erc4626/ponder.config.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      chainId: 1,
+      transport: http(process.env.PONDER_RPC_URL_1),
+    },
+  },
+  contracts: {
+    ERC4626: {
+      network: "mainnet",
+      abi: erc4626ABI,
+      address: "0xc21F107933612eCF5677894d45fc060767479A9b",
+      startBlock: 13142655,
```
Comment (kyscott18) [inline examples/reference-erc4626/ponder.config.ts:17]: Was Aave v2 wrappers deployed for this block range?

File: examples/reference-erc4626/ponder.config.ts (hunk 2 above)
Comment (kdonthi) [reply examples/reference-erc4626/ponder.config.ts:2]: 😄

File: examples/reference-erc4626/ponder.config.ts (hunk 3 above)
Comment (kdonthi) [reply examples/reference-erc4626/ponder.config.ts]: just fixed it, good catch

File: examples/reference-erc4626/package.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "ponder-examples-reference-erc20",
```
Comment (0xOlias) [inline examples/reference-erc4626/package.json:2]: Will need to update this to `ponder-examples-reference-erc4626` to play nice with changesets.

File: examples/reference-erc4626/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import { ponder } from "@/generated";
+
+ponder.on("ERC4626:Deposit", async ({ event, context }) => {
+  // keep assets and shares count for account
```
Comment (0xOlias) [inline examples/reference-erc4626/src/index.ts:4]: Should be removed, or a complete sentence like `Keep assets and shares count for account.`

File: pnpm-lock.yaml [hunk 6]
```diff
@@ -36,7 +36,7 @@ importers:
     devDependencies:
       '@graphprotocol/graph-cli':
         specifier: 0.61.0
-        version: 0.61.0(@types/node@20.10.3)(node-fetch@3.3.2)(typescript@5.3.3)
+        version: 0.61.0(@types/node@20.10.3)(node-fetch@3.3.2)(typescript@5.3.2)
```
Comment (0xOlias) [inline pnpm-lock.yaml:39]: These version diffs (which appear unrelated to the example app you added) should probably be removed and might cause a conflict with main. Will circle back on this.

File: examples/reference-erc4626/README.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
+# Example ERC4626 token API
```
Comment (0xOlias) [inline examples/reference-erc4626/README.md:1]: Build is failing because our build step for the `create-ponder` templates expects a `.env.example` file in each example directory. You should add one like this (the same one as `reference-erc20`).

```
# Mainnet RPC URL used for fetching blockchain data. Alchemy is recommended.
PONDER_RPC_URL_1=https://eth-mainnet.g.alchemy.com/v2/...

# (Optional) Postgres database URL. If not provided, SQLite will be used. 
DATABASE_URL=
```

**Review by 0xOlias — CHANGES_REQUESTED** [review]
